server/sonar-search/src/main/java/org/sonar/search/SearchServer.java

```diff
@@ -50,56 +57,182 @@ public SearchServer(Props props) {
+    if (buildServerPath2.toFile().exists()) {
+      return buildServerPath2.toAbsolutePath();
     }
-    node = NodeBuilder.nodeBuilder().settings(esSettings).build();
-    node.start();
+
+    throw new IllegalStateException(String.format("Cannot find elasticsearch binary %s in either %s or %s or %s or %s",
+            executable,
+            Paths.get(".").toAbsolutePath().getParent(),
+            Paths.get(assembly).toAbsolutePath(),
+            Paths.get(parentAssembly).toAbsolutePath(),
+            Paths.get(grandParentAssembly).toAbsolutePath()
+    ));
   }
 
-  // copied from https://github.com/elastic/elasticsearch/blob/v2.3.3/core/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java
-  private static void initBootstrap() {
-    // init lucene random seed. it will use /dev/urandom where available:
-    StringHelper.randomId();
+  private static String getExecutable() {
+    if (System.getProperty("os.name").startsWith("Windows")) {
+      return "elasticsearch/bin/elasticsearch.bat";
+    }
+    return "elasticsearch/bin/elasticsearch";
   }
 
+//  private void configureIndexDefaultSettings(EsSettings settings, PreBuiltTransportClient client) {
+//    Settings.Builder indexSettings = Settings.builder();
+//    settings.configureIndexDefaults(indexSettings);
+//    client.admin().indices().putTemplate(new PutIndexTemplateRequest().settings(indexSettings));
+//  }
+
   @Override
   public Status getStatus() {
-    boolean esStatus = node != null && node.client().admin().cluster().prepareHealth()
-        .setWaitForYellowStatus()
-        .setTimeout(TimeValue.timeValueSeconds(30L))
-        .get()
-        .getStatus() != ClusterHealthStatus.RED;
-    if (esStatus) {
-      return Status.OPERATIONAL;
+    Status status = null;
+    try {
+      status = getStatus2();
+      System.out.println("ES STATUS "+status);
+      return status;
+    } catch (Exception e) {
+      System.out.println("ES STATUS "+e.getMessage());
+      throw new RuntimeException(e);
+    }
+  }
+
+  private Status getStatus2() {
+    String urlString = url+"/_cluster/health";
+    try {
+      URL url = new URL(urlString);
+      URLConnection urlConnection = url.openConnection();
+      InputStream inputStream = urlConnection.getInputStream();
+      String line;
+      BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+      while ((line = reader.readLine()) != null) {
+        if (line.contains("\"status\"")) {
+          if (line.contains("\"red\"")) {
+            return Status.DOWN;
+          }
+          if (line.contains("\"yellow\"")) {
+            return Status.OPERATIONAL;
+          }
+          if (line.contains("\"green\"")) {
+            return Status.OPERATIONAL;
+          }
+        }
+      }
+    } catch (MalformedURLException e) {
+      throw new RuntimeException(String.format("Cannot contact local Elasticsearch instance via url '%s'", urlString), e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00112)